doc: document security issues with url.parse() (#34226)
nodejs/node: doc/api/url.md

A `URIError` is thrown if the `auth` property is present but cannot be decoded.

> Use of the legacy `url.parse()` method is not recommended. All users should

Reviewer: ```suggestion
Use of the legacy `url.parse()` method is discouraged. Users should
```

> migrate to the WHATWG `URL` api. Because the `url.parse()` method uses a

Reviewer: ```suggestion
use the WHATWG `URL` API. Because the `url.parse()` method uses a
```

> lenient, non-standards compliant algorithm for parsing URL strings, security

Reviewer 2: ```suggestion
lenient, non-standard algorithm for parsing URL strings, security
```
Take-it-or-leave-it suggestion but I expect this'll be a little easier to read for non-native speakers.

> issues can be introduced. Specifically, issues with [hostname spoofing][] and

Reviewer: ```suggestion
issues can be introduced. Specifically, issues with [host name spoofing][] and
```

> incorrect handling of user info (usernames and passwords) have been identified.

Reviewer: ```suggestion
incorrect handling of usernames and passwords) have been identified.
```

[WHATWG URL Standard]: https://url.spec.whatwg.org/
[WHATWG URL]: #url_the_whatwg_url_api
[examples of parsed URLs]: https://url.spec.whatwg.org/#example-url-parsing

> [hostname spoofing]: https://hackerone.com/reports/678487

Reviewer: ```suggestion
[host name spoofing]: https://hackerone.com/reports/678487
```
